src/identifier.js

```diff
@@ -47,9 +47,9 @@ Identifier.prototype.lookup_value = function (parent) {
   case '$data': return $data;
   default:
   }
-  if (token in $data) { return $data[token]; }
-  if (token in $context) { return $context[token]; }
-  if (token in globals) { return globals[token]; }
+  if (typeof $data == "object" && token in $data) { return $data[token]; }
```

Comment:
In what scenario is this a problem?

If `$data` is null or undefined, then it will be set to `{}` on line 36.  Same for `$globals`.  Every other value should be ok here.

The problem with testing for `object` typeof is that it ignores `function` types, which are perfectly valid for `$data`. There may be others e.g. Symbols, string/number (subclassed), etc.  We could use a try/catch, but that could slow down the interpreter substantially.

`$context` is an object by definition, so  we shouldn't need a test for that in the ordinary case.